icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java

```diff
@@ -361,14 +361,18 @@ public TestData(String source, String target, double conversionRate, double offs
     @Test
     public void testConverter() {
         class TestData {
+            String sourceIdentifier;
+            String targetIdentifier;
             MeasureUnitImpl source;
             MeasureUnitImpl target;
             BigDecimal input;
             BigDecimal expected;
 
-            TestData(String source, String target, double input, double expected) {
-                this.source = MeasureUnitImpl.UnitsParser.parseForIdentifier(source);
-                this.target = MeasureUnitImpl.UnitsParser.parseForIdentifier(target);
+            TestData(String sourceIdentifier, String targetIdentifier, double input, double expected) {
+                this.sourceIdentifier = sourceIdentifier;
+                this.targetIdentifier = targetIdentifier;
+                this.source = MeasureUnitImpl.UnitsParser.parseForIdentifier(sourceIdentifier);
+                this.target = MeasureUnitImpl.UnitsParser.parseForIdentifier(targetIdentifier);
```

Comment:
Optional: I would probably have moved the MeasureUnitImpl construction (the parseForIdentifier calls) into the test body, making it more similar to C++.

Reply:
done